### PR TITLE
create student, teacher classes Inherits from Person class

### PR DESCRIPTION
**Pull request by naji-97**
- [ ] Create class Person
- [ ] Instance vars: @id, @name, and @age.
- [ ] Constructor with name, age, and parent_permission as parameter. name and parent_permission are optional and have default values of "Unknown" and true.
- [ ] Getters for @id, @name, and @age.
- [ ] Setters for @name and @age.
- [ ] Private method of_age? that returns true if @age is greater or equal to 18 and false otherwise.
- [ ] Public method can_use_services? that returns true if person is of age or if they have permission from parents.
- [ ] Create class Student Inherits from Person.
- [ ] Constructor extends parent's constructor by adding @classroom and a parameter for it.
- [ ] Method play_hooky that returns "¯\(ツ)/¯".
- [ ] Create class Teacher Inherits from Person.
- [ ] Constructor extends parent's constructor by adding @specialization and a parameter for it.
- [ ] Override can_use_services? so it always returns true.

### CODE REVIEW

**Review by frank1738 — COMMENTED** [review]
Hi @naji-97 , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

- [ ] Kindly add a readme to your project for proper documentation

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    @parent_permission = parent_permission
+  end
+
+  def of_age?
+    @age >= 18
+  end
+
+  def can_use_services?
+    of_age? || @parent_permission
+  end
```
Comment (Ibizugbe) [inline person.rb:20]: Just as required in this milestone, the `of_age` method is meant to be a private method😉. Kindly do well to implement this👍.
image

```suggestion

  def can_use_services?
    of_age? || @parent_permission
  end
  
  private
  
  def of_age?
    @age >= 18
  end

```

**Review by Ibizugbe — CHANGES_REQUESTED** [review]
Hi @naji-97 👋👋👋, 

tryagain


Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights✅✅✅

- [x] Linters are passing👍.
- [x] gitflow is used👍.
- [x] `student` and `teacher` class have been properly implemented👍.


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by GzimAsani — APPROVED** [review]
# STATUS: APPROVED :100:         
## There is nothing else to say than it's time to merge... :partying_face: 
Cheers and Happy coding! :clap:  :computer:  :wine_glass: 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


## Optional suggestions


_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- 

### Please be reminded to not open a new pull request after you get a review, only use the same one :thought_balloon: 


 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**